src/senselab/audio/tasks/data_augmentation.py

```diff
@@ -1,44 +1,92 @@
 """This module implements some utilities for audio data augmentation."""
 
-from typing import Any, Dict
+from typing import Any, Dict, List, Union
 
 import torch
 from datasets import Dataset
 from torch_audiomentations import Compose
 
+from senselab.utils.data_structures.audio import (
+    Audio,
+    batch_audios,
+    unbatch_audios,
+)
+from senselab.utils.device import DeviceType, _select_device_and_dtype
 from senselab.utils.tasks.input_output import (
     _from_dict_to_hf_dataset,
     _from_hf_dataset_to_dict,
 )
 
 
-def augment_hf_dataset(
-    dataset: Dict[str, Any], augmentation: Compose, audio_column: str = "audio"
-) -> Dict[str, Any]:
+def augment_audio_dataset(
+    audios: List[Audio], augmentation: Compose, device_options: Union[DeviceType, List[DeviceType]] = [DeviceType.CPU]
+) -> List[Audio]:
+    """Augments all provided audios with a given augmentation, either individually or all batched together.
+
+    Augment all audios with a user defined augmentation that can be a composition of multiple augmentations. This
+    augmentation is either performed on each audio individually or all of the audios provided are batched together
+    and run at once. NOTE: if batching, all audios must have the same sampling rate.
+
+    Args:
+        audios: List of Audios whose data will be augmented with the given augmentations
+        augmentation: A Composition of augmentations to run on each audio (uses torch-audiomentations), should have its
+            output_type set to "dict"
+        device_options: The device, or a List of possible devices, to use for augmenting. If the chosen device
+            is MPS or CUDA then the audios are all batched together, so for optimal performance, batching should
+            be done by passing a batch_size worth of audios ar a time
+
+    Returns:
+        List of audios that has passed the all of input audios through the provided augmentation. This does
+            not necessarily mean that the augmentation has been run on every audio. For more information,
+            see the torch-audiomentations documentation.
+    """
+    augmentation.output_type = "dict"
+    new_audios = []
+    device_type, dtype = _select_device_and_dtype(
```

Comment:
how do you manage the scenario when the developer wants to use a device which is not supported? this is the question you asked me

Reply:
I dont